cli/src/main.rs

```diff
@@ -1,19 +1,561 @@
-use std::{fs::File, io::Error};
+use clap::{Parser, Subcommand};
+use log::{debug, error, info, trace, warn};
 
-use clap::Parser;
+use std::{
+    env,
+    error::Error,
+    future::Future,
+    path::{Path, PathBuf},
+    process::Stdio,
+};
+use thiserror::Error;
+use tokio::{
+    io::{AsyncBufReadExt, BufReader},
+    process::{Child, ChildStderr, ChildStdout, Command},
+    time::{Duration, Instant},
+};
 
-#[derive(Parser, Debug)]
+pub mod development {
+    tonic::include_proto!("development");
+}
+use development::development_client::DevelopmentClient;
+use development::{DeployReply, DeployRequest, EchoReply, EchoRequest, Empty};
+
+#[derive(Error, Debug)]
+pub enum ServerError {
+    #[error("development_server error")]
+    ServerError { cause: String },
+    #[error("unknown failure")]
+    Unknown,
+}
+
+#[derive(PartialEq, Debug)]
+pub enum ServerState {
+    NotStarted,
+    StartingUp,
+    Started,
+}
+
+#[derive(Error, Debug)]
+pub enum PollError {
+    #[error("not yet started")]
+    NotStarted,
+    #[error("already started")]
+    Started,
+    #[error("development_server error")]
```

Comment:
Check out -> info!("Received Echo. message: {:?}", message);

We can do the following to produce some awesome errors

```rust
#[error("development_server error {cause:?}")]
```

I recently had to go back in my code to add this in.